Hierarchical data

ahie opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/8842)
<!-- Reviewable:end -->

Commit:
Initial HierarchicalDataProvider for TreeGrid

ahie committed:
Add initial in-memory hierarchical data implementation

Commit:
Cleanup and javadocs for HierarchicalDataProvider

Commit:
Do not drop expanded node keys

Cannot drop those since needed for fetching items when scrolling.

ahie committed:
Add configurable filtering to InMemoryHierarchicalDataProvider

Implement removal of items in HierarchyData

Add javadocs to hierarchical data APIs

Commit:
Fixed bug when scrolling tree grid up

Added convenience methods for TreeGridElement for testing.

ahie committed:
Add fixes to InMemoryHierarchicalDataProvider, tests

ahie committed:
Fix exclude in ClassesSerializableTest

ahie committed:
Fix incorrect class in TreeGridBasicFeatures

pleku commented:
Reviewed 4 of 19 files at r1, 2 of 11 files at r2, 2 of 6 files at r4, 1 of 7 files at r5, 3 of 6 files at r6.
Review status: 12 of 31 files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java, line 165 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC0NWaOKC1zwwmTMVk:-KfC0NWbX-0RfAOvo4JR:b-i7uiaj) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java#L165)):*
> ```Java
>      * @param oldRowData
>      *            the previous row data
>      */
> ```

Missing `@since`

---

*[client/src/main/java/com/vaadin/client/connectors/data/HierarchicalDataCommunicatorConnector.java, line 43 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC0fmQvmtIRtaK5rVt:-KfC0fmQvmtIRtaK5rVu:bxtn6cl) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/client/src/main/java/com/vaadin/client/connectors/data/HierarchicalDataCommunicatorConnector.java#L43)):*
> ```Java
> 
>         /*
>          * Since server side can't know the index a random item, any
> ```

..the index of a random...

---

*[server/src/main/java/com/vaadin/data/HierarchyData.java, line 208 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC2aiwAPcHmbtTVfW7:-KfC2aiwAPcHmbtTVfW8:b-abb3w1) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/HierarchyData.java#L208)):*
> ```Java
>      * 
>      * @param item
>      *            the item to remove
> ```

What about `null` item ?

---

*[server/src/main/java/com/vaadin/data/HierarchyData.java, line 229 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC2X34YKC9d5qcJnN8:-KfC2X35hqgzTTZJ7HnB:bp96ycv) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/HierarchyData.java#L229)):*
> ```Java
>      * 
>      * @param item
>      *            the item for which to retrieve child items for
> ```

what about `null` item ?

---

*[server/src/main/java/com/vaadin/data/provider/AbstractHierarchicalDataProvider.java, line 26 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC3WcfNWgnm2l60kdr:-KfC3WcfNWgnm2l60kds:b8emmrs) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/provider/AbstractHierarchicalDataProvider.java#L26)):*
> ```Java
> 
> /**
>  *
> ```

Javadocs missing

---

*[server/src/main/java/com/vaadin/data/provider/InMemoryHierarchicalDataProvider.java, line 221 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC3zX2efpCl6N517YW:-KfC3zX2efpCl6N517YX:b-d0wxhj) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/provider/InMemoryHierarchicalDataProvider.java#L221)):*
> ```Java
>         return queryFilter.map(stream::filter).orElse(stream);
>     }
> }
> ```

new line

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 69 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC54N1TFpySzH8I7uz:-KfC54N1TFpySzH8I7v-:bzaihsw) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L69)):*
> ```Java
> 
>     @Override
>     public void setItems(Collection<T> items) {
> ```

Please override javadocs

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 75 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC55ojR3JYOIAxV_1_:-KfC55ojR3JYOIAxV_1a:bzaihsw) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L75)):*
> ```Java
> 
>     @Override
>     public void setItems(Stream<T> items) {
> ```

Please override javadocs

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 81 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC571ANj8AV41xuM4F:-KfC571ANj8AV41xuM4G:bzaihsw) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L81)):*
> ```Java
> 
>     @Override
>     public void setItems(@SuppressWarnings("unchecked") T... items) {
> ```

Please override javadocs

---

*[uitest/src/main/java/com/vaadin/tests/components/treegrid/InMemoryHierarchicalDataProvider.java, line 20 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC6z-TY19ipIZJYtFZ:-KfC6z-UBT0iJp8GvwO-:b-okqsxq) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/uitest/src/main/java/com/vaadin/tests/components/treegrid/InMemoryHierarchicalDataProvider.java#L20)):*
> ```Java
> import com.vaadin.tests.components.treegrid.TreeGridBasicFeatures.HierarchicalTestBean;
> 
> class InMemoryHierarchicalDataProvider
> ```

Should be removed, and if not already so, replaced by the one from  _vaadin server_

---

*[uitest/src/main/java/com/vaadin/tests/components/treegrid/LazyHierarchicalDataProvider.java, line 15 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC6PyNMJh9-B-stueu:-KfC6PyNMJh9-B-stuev:b-k5zjrc) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/uitest/src/main/java/com/vaadin/tests/components/treegrid/LazyHierarchicalDataProvider.java#L15)):*
> ```Java
> 
> public class LazyHierarchicalDataProvider
>         implements HierarchicalDataProvider<HierarchicalTestBean, Void> {
> ```

Should extend `AbstractHierarchicalDataProvider`

---

*[uitest/src/main/java/com/vaadin/tests/components/treegrid/LazyHierarchicalDataProvider.java, line 81 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC6Nhww8OqdqaCWvYd:-KfC6Nhww8OqdqaCWvYe:b-l6mwxj) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/uitest/src/main/java/com/vaadin/tests/components/treegrid/LazyHierarchicalDataProvider.java#L81)):*
> ```Java
>     private boolean internalHasChildren(HierarchicalTestBean node) {
>         return node.getDepth() < depth;
>         // return node.getDepth() < depth && node.getIndex() % 2 == 1;
> ```

remove

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Review status: 12 of 31 files reviewed at latest revision, 17 unresolved discussions, some commit checks failed.

---

*[client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java, line 88 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCQe4dtIeOr2rNtXJM:-KfCQe4dtIeOr2rNtXJN:b-uv5b8x) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java#L88)):*
> ```Java
> 
>                         @Override
>                         public void removeRowData(int firstRowIndex,
> ```

This method should be renamed to "removeRows" to be consistent with the other one.

---

*[client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java, line 165 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC0NWaOKC1zwwmTMVk:-KfCQw7FViyJNK3vAKXl:b6srdec) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java#L165)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Missing `@since`
</blockquote></details>

IMO this is something that we could use to provide partial data updates. similar approach is something I've been pondering since 7.5.. I'll come up with a more complete plan for future versions.

---

*[server/src/main/java/com/vaadin/data/HierarchyData.java, line 219 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCSAddXjEcIVjTxbJy:-KfCSAddXjEcIVjTxbJz:b-4tcgkw) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/HierarchyData.java#L219)):*
> ```Java
>                     "Item '" + item + "' not in the hierarchy");
>         }
>         getChildren(item).forEach(child -> removeItem(child));
> ```

This piece of code looks like it will explode with a nice ConcurrentModificationException when you actually start removing the children. Is there a test for removing a root node?

---

*[server/src/main/java/com/vaadin/data/provider/AbstractHierarchicalDataProvider.java, line 33 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCT0UTP7HO53X7pvA8:-KfCT0UTP7HO53X7pvA9:bjr5enl) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/provider/AbstractHierarchicalDataProvider.java#L33)):*
> ```Java
>  * @param <F>
>  */
> public abstract class AbstractHierarchicalDataProvider<T, F>
> ```

This class can't extend the regular AbstractDataProvider?

---

*[server/src/main/java/com/vaadin/data/provider/HierarchicalDataCommunicator.java, line 82 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCTOZ9CB5C7k6wkb22:-KfCTOZ9CB5C7k6wkb23:bszacy5) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/provider/HierarchicalDataCommunicator.java#L82)):*
> ```Java
> 
>     @Override
>     public void beforeClientResponse(boolean initial) {
> ```

This method content in the super class should be split into pieces and made to be overridden. not calling the super at all means that if there is some functionality in a base class of the base class, it won't be executed.

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 57 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCU2aVkpsp4mQ97KuA:-KfCU2aVkpsp4mQ97KuB:b-k8b9qq) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L57)):*
> ```Java
>         registerRpc(new NodeCollapseRpc() {
>             @Override
>             public void toggleCollapse(String rowKey, int rowIndex,
> ```

Nit: This is actually not "toggle" anymore, but more of a "setNodeCollapsed"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

ahie committed:
Add fixes to hierarchical data

ahie committed:
Parameterize TreeGridBasicFeaturesTest

ahie commented:
Review status: 4 of 31 files reviewed at latest revision, 16 unresolved discussions, some commit checks failed.

---

*[client/src/main/java/com/vaadin/client/connectors/data/HierarchicalDataCommunicatorConnector.java, line 43 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC0fmQvmtIRtaK5rVt:-KfFpDKryegLGdgbZWJ_:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/client/src/main/java/com/vaadin/client/connectors/data/HierarchicalDataCommunicatorConnector.java#L43)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

..the index of a random...
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/data/HierarchyData.java, line 208 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC2aiwAPcHmbtTVfW7:-KfFpE8Ju7rt8juYQCXS:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/HierarchyData.java#L208)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

What about `null` item ?
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/data/HierarchyData.java, line 219 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCSAddXjEcIVjTxbJy:-KfFpF-Ere6Tv7lyv9D1:b-pms4ot) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/HierarchyData.java#L219)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

This piece of code looks like it will explode with a nice ConcurrentModificationException when you actually start removing the children. Is there a test for removing a root node?
</blockquote></details>

Fixed and added a test.

---

*[server/src/main/java/com/vaadin/data/HierarchyData.java, line 229 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC2X34YKC9d5qcJnN8:-KfFpPKw2p2AYnP_h3Nz:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/HierarchyData.java#L229)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

what about `null` item ?
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/data/provider/AbstractHierarchicalDataProvider.java, line 26 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC3WcfNWgnm2l60kdr:-KfFpQ9a0HAdN5kOOm39:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/provider/AbstractHierarchicalDataProvider.java#L26)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Javadocs missing
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/data/provider/InMemoryHierarchicalDataProvider.java, line 221 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC3zX2efpCl6N517YW:-KfFpVZKuizcb6sFoiUR:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/provider/InMemoryHierarchicalDataProvider.java#L221)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

new line
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 69 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC54N1TFpySzH8I7uz:-KfFpX9mrI3DOgcgxKNR:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L69)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Please override javadocs
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 75 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC55ojR3JYOIAxV_1_:-KfFpXaXCUofV05nPPID:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L75)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Please override javadocs
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 81 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC571ANj8AV41xuM4F:-KfFpY-BSlbcwLUfZ78W:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L81)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Please override javadocs
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/components/treegrid/LazyHierarchicalDataProvider.java, line 15 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC6PyNMJh9-B-stueu:-KfFpZqgyE4Rv6JIgN2n:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/uitest/src/main/java/com/vaadin/tests/components/treegrid/LazyHierarchicalDataProvider.java#L15)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Should extend `AbstractHierarchicalDataProvider`
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/components/treegrid/LazyHierarchicalDataProvider.java, line 81 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC6Nhww8OqdqaCWvYd:-KfFp_VpgHkadplVX87_:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/uitest/src/main/java/com/vaadin/tests/components/treegrid/LazyHierarchicalDataProvider.java#L81)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

remove
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/components/treegrid/InMemoryHierarchicalDataProvider.java, line 20 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfC6z-TY19ipIZJYtFZ:-KfFpi16m0Z8u_rglK8p:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/uitest/src/main/java/com/vaadin/tests/components/treegrid/InMemoryHierarchicalDataProvider.java#L20)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Should be removed, and if not already so, replaced by the one from  _vaadin server_
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

ahie committed:
Merge remote-tracking branch 'github/master' into hierarchical_data

pleku commented:
Reviewed 1 of 19 files at r1, 1 of 11 files at r2, 1 of 7 files at r5, 11 of 14 files at r7.
Review status: 16 of 30 files reviewed at latest revision, 11 unresolved discussions.

---

*[server/src/main/java/com/vaadin/data/HierarchyData.java, line 208 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFu45RN0DcCCH4kfRq:-KfFu45RN0DcCCH4kfRr:b9427vd) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/data/HierarchyData.java#L208)):*
> ```Java
>      * 
>      * @param item
>      *            the item to remove, or null to clear all data
> ```

Might be a handy shorthand to clear() all data 

---

*[server/src/main/java/com/vaadin/data/provider/DataCommunicator.java, line 242 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFuPkEH6yKPtHdI4uX:-KfFuPkFcxaxn1FwwS11:b-po6mmc) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/data/provider/DataCommunicator.java#L242)):*
> ```Java
>     }
> 
>     protected Object getFilter() {
> ```

Missing javadocs

---

*[server/src/main/java/com/vaadin/data/provider/DataCommunicator.java, line 246 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFuQWp36g2-JdRJgqp:-KfFuQWp36g2-JdRJgqq:b-po6mmc) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/data/provider/DataCommunicator.java#L246)):*
> ```Java
>     }
> 
>     protected DataCommunicatorClientRpc getClientRpc() {
> ```

Missing javadocs

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 88 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFw58DWAvf9ousYjhi:-KfFw58DWAvf9ousYjhj:b-qpvdnw) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L88)):*
> The provided

The returned HierachyData

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 90 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFvnPLPDa54Pq2dnTu:-KfFvnPLPDa54Pq2dnTv:b2benrr) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L90)):*
> ```Java
>      * The provided HierarchyData instance may be used as-is. Subsequent
>      * modification of the object might cause inconsistent data to be shown in
>      * the component unless it is explicitly instructed to read the data again.
> ```

#ffs this last sentence please.

Meaning, answer modifications on _what_ should cause you do to _what_ in order to get _what_ show the updated data. 

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 124 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFwa20GMaxLIH2cL7p:-KfFwa20GMaxLIH2cL7q:bwfswmv) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L124)):*
> ```Java
>      * The provided HierarchyData instance may be used as-is. Subsequent
>      * modification of the object might cause inconsistent data to be shown in
>      * the component unless it is explicitly instructed to read the data again.
> ```

Same #ffs as previously

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 158 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFweUu6EBIMJhq9O2Y:-KfFweUu6EBIMJhq9O2Z:bwfswmv) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L158)):*
> ```Java
>      * The provided HierarchyData instance may be used as-is. Subsequent
>      * modification of the object might cause inconsistent data to be shown in
>      * the component unless it is explicitly instructed to read the data again.
> ```

Same #ffs as previously

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

Commit:
Refactor DataConnector data sending to client

pleku commented:
Review status: 15 of 30 files reviewed at latest revision, 10 unresolved discussions.

---

*[server/src/main/java/com/vaadin/data/provider/HierarchicalDataCommunicator.java, line 82 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCTOZ9CB5C7k6wkb22:-KfG7yquLB7jFSRLAEdi:bb6un1j) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/provider/HierarchicalDataCommunicator.java#L82)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

This method content in the super class should be split into pieces and made to be overridden. not calling the super at all means that if there is some functionality in a base class of the base class, it won't be executed.
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

ahie commented:
Review status: 10 of 30 files reviewed at latest revision, 12 unresolved discussions.

---

*[client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java, line 88 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCQe4dtIeOr2rNtXJM:-KfFyfaHnIdlvLDWnPLk:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java#L88)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

This method should be renamed to "removeRows" to be consistent with the other one.
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/data/HierarchyData.java, line 208 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFu45RN0DcCCH4kfRq:-KfFz6mpwWN6lb3RNjL5:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/data/HierarchyData.java#L208)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Might be a handy shorthand to clear() all data 
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/data/provider/AbstractHierarchicalDataProvider.java, line 33 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCT0UTP7HO53X7pvA8:-KfFzFYLwm7An5OGUNRq:b-bmr1f4) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/provider/AbstractHierarchicalDataProvider.java#L33)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

This class can't extend the regular AbstractDataProvider?
</blockquote></details>

Done. Should probably also take a closer look if it's feasible to have `InMemoryHierarchicalDataProvider` share code with `ListDataProvider` without breaking backwards compatibility with 8.0, while at the same time having the class hierarchy make sense.

---

*[server/src/main/java/com/vaadin/data/provider/DataCommunicator.java, line 242 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFuPkEH6yKPtHdI4uX:-KfG5YaKqzm43sJfsid5:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/data/provider/DataCommunicator.java#L242)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Missing javadocs
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/data/provider/DataCommunicator.java, line 246 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFuQWp36g2-JdRJgqp:-KfG4SIbRPej8eY-RWwL:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/data/provider/DataCommunicator.java#L246)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Missing javadocs
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 57 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCU2aVkpsp4mQ97KuA:-KfG65Doe278Ad_aNK5q:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L57)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

Nit: This is actually not "toggle" anymore, but more of a "setNodeCollapsed"
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 88 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFw58DWAvf9ousYjhi:-KfGJxjRQh9dK53xS4re:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L88)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

> The provided

The returned HierachyData
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 90 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFvnPLPDa54Pq2dnTu:-KfGJy8d6ZJhzADlwNqc:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L90)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

#ffs this last sentence please.

Meaning, answer modifications on _what_ should cause you do to _what_ in order to get _what_ show the updated data. 
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 124 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFwa20GMaxLIH2cL7p:-KfGJyZzK1wflaEpnMMh:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L124)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Same #ffs as previously
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/ui/TreeGrid.java, line 158 at r7](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfFweUu6EBIMJhq9O2Y:-KfGJz93FChnNs8luHfd:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c1b4ca0b3fc8d82a3dd8175190defe7ffc93b71b/server/src/main/java/com/vaadin/ui/TreeGrid.java#L158)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Same #ffs as previously
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

ahie committed:
Refactor AbstractHierarchicalDataProvider, fix javadocs, add HierarchyData#clear()

pleku commented:
Reviewed 1 of 19 files at r1, 1 of 11 files at r2, 1 of 6 files at r6, 1 of 14 files at r7, 9 of 10 files at r9.
Review status: 23 of 30 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

ahie committed:
Improve HasItems javadocs

ahie committed:
Merge remote-tracking branch 'github/master' into hierarchical_data

ahie committed:
Revert abstract from GridDeclarativeTest

tsuoanttila commented:
Reviewed 1 of 19 files at r1, 1 of 5 files at r3, 1 of 6 files at r4, 2 of 7 files at r5, 2 of 10 files at r9, 1 of 5 files at r10.
Review status: 26 of 30 files reviewed at latest revision, 6 unresolved discussions.

---

*[client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java, line 90 at r9](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfGh7Irf8MrBOdfo8JB:-KfGh7Irf8MrBOdfo8JC:b-zcsvmf) ([raw file](https://github.com/vaadin/framework/blob/9376b14202dd51525109dddbdc7da4a779b9aa44/client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java#L90)):*
> ```Java
>                         public void removeRows(int firstRowIndex,
>                                 int count) {
>                             VaadinDataSource.this.removeRowData(firstRowIndex,
> ```

Unnecessary VaadinDataSource.this.

---

*[server/src/main/java/com/vaadin/data/provider/AbstractHierarchicalDataProvider.java, line 33 at r6](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfCT0UTP7HO53X7pvA8:-KfGy9bnX5SEy8KhkkYc:b-1n25cs) ([raw file](https://github.com/vaadin/framework/blob/b8a5933e0ccb1331b34869cb82dbfc89fd703a1b/server/src/main/java/com/vaadin/data/provider/AbstractHierarchicalDataProvider.java#L33)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

Done. Should probably also take a closer look if it's feasible to have `InMemoryHierarchicalDataProvider` share code with `ListDataProvider` without breaking backwards compatibility with 8.0, while at the same time having the class hierarchy make sense.
</blockquote></details>

Seeing that content in this class is basically empty, I think we could make `InMemoryHierarchicalDataProvider extends ListDataProvider implements HierarchicalDataProvider`. To be verfied. Can also be it's own patch.

---

*[server/src/main/java/com/vaadin/data/provider/HierarchyMapper.java, line 56 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfGyy3AHrj9VX9lOOyi:-KfGyy3AHrj9VX9lOOyj:b-wkdilk) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/server/src/main/java/com/vaadin/data/provider/HierarchyMapper.java#L56)):*
> ```Java
>          * key.
>          */
>         final TreeNode node;
> ```

Package visibility on purpose? Also I might need to re-review this file with thought.

---

*[server/src/test/java/com/vaadin/data/provider/HierarchyMapperTest.java, line 11 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfGzpXz2RFeeGlgJG0E:-KfGzpXz2RFeeGlgJG0F:b-k5ifdv) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/server/src/test/java/com/vaadin/data/provider/HierarchyMapperTest.java#L11)):*
> ```Java
> import com.vaadin.data.provider.HierarchyMapper.TreeNode;
> 
> public class HierarchyMapperTest {
> ```

This file could use a couple of clarifying comment on what is the internal state of the mapper supposed to be in each logical step.

---

*[uitest/src/main/java/com/vaadin/tests/components/table/ColumnWidthsAfterChangeTableColumnsCountOrOrder.java, line 105 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfH-G2-ApYUZkmHe4cU:-KfH-G2-ApYUZkmHe4cV:b-oz63q2) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/uitest/src/main/java/com/vaadin/tests/components/table/ColumnWidthsAfterChangeTableColumnsCountOrOrder.java#L105)):*
> ```Java
>         @Override
>         public String toString() {
>             return "HierarchicalTestBean{" + "id=" + id + ", name='" + name + '\''
> ```

What is the purpose of this change?

---

*[uitest/src/test/java/com/vaadin/tests/components/treegrid/TreeGridScrollingTest.java, line 12 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfH-N0fSxmNIFtKrgOt:-KfH-N0fSxmNIFtKrgOu:bkndupl) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/uitest/src/test/java/com/vaadin/tests/components/treegrid/TreeGridScrollingTest.java#L12)):*
> ```Java
> 
>     @Test
>     public void testScrollingTree_expandCollapseFromBeginning_correctItemsShown() {
> ```

This test seems really long. Can it be split into logical pieces that test various parts rather than testing everything in one go?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

Commit:
Small review fixes

pleku commented:
Reviewed 1 of 5 files at r10.
Review status: 24 of 29 files reviewed at latest revision, 5 unresolved discussions.

---

*[client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java, line 90 at r9](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfGh7Irf8MrBOdfo8JB:-KfH1afi7I3MLByzm-Wf:bb6un1j) ([raw file](https://github.com/vaadin/framework/blob/9376b14202dd51525109dddbdc7da4a779b9aa44/client/src/main/java/com/vaadin/client/connectors/data/DataCommunicatorConnector.java#L90)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

Unnecessary VaadinDataSource.this.
</blockquote></details>

Done

---

*[server/src/main/java/com/vaadin/data/provider/HierarchyMapper.java, line 56 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfGyy3AHrj9VX9lOOyi:-KfH1pqxq-fUDqcHFzhU:bflwga2) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/server/src/main/java/com/vaadin/data/provider/HierarchyMapper.java#L56)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

Package visibility on purpose? Also I might need to re-review this file with thought.
</blockquote></details>

Yes, used by `HierarchicalDataCommunicator`. Could add the getters, but haven't since that just felt boilerplaty

---

*[server/src/test/java/com/vaadin/data/provider/HierarchyMapperTest.java, line 11 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfGzpXz2RFeeGlgJG0E:-KfH2SN6ELyv0-M3FNsl:b4wfsnj) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/server/src/test/java/com/vaadin/data/provider/HierarchyMapperTest.java#L11)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

This file could use a couple of clarifying comment on what is the internal state of the mapper supposed to be in each logical step.
</blockquote></details>

Not sure what type of comments would be clarifying enough, to me the `verifyX`methods state what it is checking, expand / collapse methods introduce new nodes

---

*[uitest/src/test/java/com/vaadin/tests/components/treegrid/TreeGridScrollingTest.java, line 12 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfH-N0fSxmNIFtKrgOt:-KfH0xP9C-fVgFOaoylZ:b-rr7woa) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/uitest/src/test/java/com/vaadin/tests/components/treegrid/TreeGridScrollingTest.java#L12)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

This test seems really long. Can it be split into logical pieces that test various parts rather than testing everything in one go?
</blockquote></details>

It is by design very long, and should not be split since testing single pieces doesn't guarantee that all the pieces work when used chained. 

---

*[uitest/src/main/java/com/vaadin/tests/components/table/ColumnWidthsAfterChangeTableColumnsCountOrOrder.java, line 105 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfH-G2-ApYUZkmHe4cU:-KfH0sbSqs7v0kndZpgR:bn4xa08) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/uitest/src/main/java/com/vaadin/tests/components/table/ColumnWidthsAfterChangeTableColumnsCountOrOrder.java#L105)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

What is the purpose of this change?
</blockquote></details>

I think it is due to eclipse rename automagic, somehow missed this

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Review status: 24 of 29 files reviewed at latest revision, 4 unresolved discussions.

---

*[server/src/main/java/com/vaadin/data/provider/HierarchyMapper.java, line 56 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfGyy3AHrj9VX9lOOyi:-KfH6PlmYoklq90tHshS:b-jpqkna) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/server/src/main/java/com/vaadin/data/provider/HierarchyMapper.java#L56)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Yes, used by `HierarchicalDataCommunicator`. Could add the getters, but haven't since that just felt boilerplaty
</blockquote></details>

Is there a reason for someone to be able to access this class in any way by extending or otherwise? if not, then make the whole class available only in the package visibility scope and document to be "Framework internal". 

---

*[server/src/test/java/com/vaadin/data/provider/HierarchyMapperTest.java, line 11 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfGzpXz2RFeeGlgJG0E:-KfH6sehNES-dxhn3nAX:b-tdks9a) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/server/src/test/java/com/vaadin/data/provider/HierarchyMapperTest.java#L11)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Not sure what type of comments would be clarifying enough, to me the `verifyX`methods state what it is checking, expand / collapse methods introduce new nodes
</blockquote></details>

At least for me, I can't see from the test method where the data is created, which makes following the test flow quite hard.

---

*[uitest/src/test/java/com/vaadin/tests/components/treegrid/TreeGridScrollingTest.java, line 12 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfH-N0fSxmNIFtKrgOt:-KfH7F1gcgEEo1gfo_Wl:b-pw92hw) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/uitest/src/test/java/com/vaadin/tests/components/treegrid/TreeGridScrollingTest.java#L12)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

It is by design very long, and should not be split since testing single pieces doesn't guarantee that all the pieces work when used chained. 
</blockquote></details>

so can you at least split it into sensibly sized methods with descriptive names?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

Commit:
Reduced API visibility, commented tests

pleku commented:
Review status: 24 of 29 files reviewed at latest revision, 4 unresolved discussions.

---

*[server/src/test/java/com/vaadin/data/provider/HierarchyMapperTest.java, line 11 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfGzpXz2RFeeGlgJG0E:-KfH_8sS1oPhCHOlS7Pc:b4fazg1) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/server/src/test/java/com/vaadin/data/provider/HierarchyMapperTest.java#L11)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

At least for me, I can't see from the test method where the data is created, which makes following the test flow quite hard.
</blockquote></details>

No data created, this just tests that each collapse / expand just updates the nodes and their indexes accordingly

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

pleku commented:
Review status: 23 of 29 files reviewed at latest revision, 4 unresolved discussions.

---

*[server/src/main/java/com/vaadin/data/provider/HierarchyMapper.java, line 56 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfGyy3AHrj9VX9lOOyi:-KfHb-Qb2XHiadVUrSEt:bb6un1j) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/server/src/main/java/com/vaadin/data/provider/HierarchyMapper.java#L56)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

Is there a reason for someone to be able to access this class in any way by extending or otherwise? if not, then make the whole class available only in the package visibility scope and document to be "Framework internal". 
</blockquote></details>

Done

---

*[uitest/src/test/java/com/vaadin/tests/components/treegrid/TreeGridScrollingTest.java, line 12 at r10](https://reviewable.io:443/reviews/vaadin/framework/8842#-KfH-N0fSxmNIFtKrgOt:-KfHau1rKl2IhFOuTp1G:bijc0un) ([raw file](https://github.com/vaadin/framework/blob/691d45e12fde09c5c3775cb52c007360b045c541/uitest/src/test/java/com/vaadin/tests/components/treegrid/TreeGridScrollingTest.java#L12)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

so can you at least split it into sensibly sized methods with descriptive names?
</blockquote></details>

Added comment clarifying what would be the proper ™ fix for this

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Reviewed 1 of 5 files at r11.
Review status: 24 of 29 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Reviewed 1 of 5 files at r11.
Review status: 25 of 29 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Reviewed 2 of 5 files at r10, 3 of 5 files at r11.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8842)*
<!-- Sent from Reviewable.io -->
